src/prognostic_equations/advection.jl

```diff
@@ -422,6 +427,284 @@ function edmfx_sgs_vertical_advection_tendency!(
+                    TD.gas_constant_air(thp, ᶜtsʲs.:($$j)) *
+                    TD.air_temperature(thp, ᶜtsʲs.:($$j)) +
+                    ᶜΦ
+                ),
+            ))
+            ᶜmseₛ = (@. lazy(
+                Y.c.sgsʲs.:($$j).q_sno * (
+                    TD.internal_energy_ice(thp, ᶜtsʲs.:($$j)) +
+                    TD.gas_constant_air(thp, ᶜtsʲs.:($$j)) *
+                    TD.air_temperature(thp, ᶜtsʲs.:($$j)) +
+                    ᶜΦ
+                ),
+            ))
+
+            # RICO works well but we are missing mse * q_ advection term
+            va = vertical_advection(ᶠwₗ³ʲs, ᶜmseₗ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            va = vertical_advection(ᶠwᵢ³ʲs, ᶜmseᵢ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            va = vertical_advection(ᶠwᵣ³ʲs, ᶜmseᵣ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            va = vertical_advection(ᶠwₛ³ʲs, ᶜmseₛ, edmfx_upwinding)
+            @. Yₜ.c.sgsʲs.:($$j).mse += va
+            #=
+                        va = vertical_advection(ᶠwₗ³ʲs, Y.c.sgsʲs.:($j).q_liq, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+                        va = vertical_advection(ᶠwᵢ³ʲs, Y.c.sgsʲs.:($j).q_ice, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+                        va = vertical_advection(ᶠwᵣ³ʲs, Y.c.sgsʲs.:($j).q_rai, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+                        va = vertical_advection(ᶠwₛ³ʲs, Y.c.sgsʲs.:($j).q_sno, edmfx_upwinding)
+                        @. Yₜ.c.sgsʲs.:($$j).mse -= Y.c.sgsʲs.:($$j).mse * va
+            =#
+
+            # mse, q_tot and moisture tracers terms proportional to 1/ρ̂ ∂zρ̂
+            ᶜinv_ρ̂ = (@. lazy(
+                specific(
+                    FT(1),
+                    Y.c.sgsʲs.:($$j).ρa,
+                    FT(0),
+                    Y.c.ρ,
+                    turbconv_model,
+                ),
+            ))
+            ᶜ∂ρ̂∂zₗ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwₗʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+            ᶜ∂ρ̂∂zᵢ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwᵢʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+            ᶜ∂ρ̂∂zᵣ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwᵣʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
+            ᶜ∂ρ̂∂zₛ = (@. lazy(
+                upwind_biased_grad(
+                    -1 * Geometry.WVector(ᶜwₛʲs.:($$j)),
+                    Y.c.sgsʲs.:($$j).ρa,
+                ),
+            ))
```

Comment:
I think all these ᶜ∂ρ̂∂z should be the same if all the ws have the same sign.

Reply:
I wonder if I should just put there `-1` and be done with it. We are assuming that sedimentation velocity only goes down. This would be making more sense if we were summing the updraft velocity with the sedimentation components.